lib/modules/dosomething/dosomething_reportback/dosomething_reportback.reportback_transformer.inc

```diff
@@ -44,6 +120,48 @@ abstract class ReportbackTransformer {
 
 
   /**
+   * Get metadata for pagination of response data.
+   *
+   * @param $parameters
+   * @param $count
+   * @param $endpoint
+   *
+   * @return array
+   */
+  protected function paginate($parameters, $endpoint) {
+    $data = array();
+
+    $data['total'] = $this->countItems($parameters);
+    $data['per_page'] = $parameters['count'];
+    $data['current_page'] = (isset($parameters['page']) && $parameters['page'] > 0) ? (int) $parameters['page'] : 1;
+    $data['total_pages'] = ceil($data['total'] / $data['per_page']);
+
+    $prevUri = $this->getPrevPageUri($data, $parameters, $endpoint);
+    $nextUri = $this->getNextPageUri($data, $parameters, $endpoint);
+
+    $data['prev_uri'] = $prevUri;
+    $data['next_uri'] = $nextUri;
+
+    return $data;
+  }
+
+
+  /**
+   * Get the offset for requests based on specified page number and count of items.
+   *
+   * @param $page
+   * @param $count
+   *
+   * @return int
+   */
+  protected function setOffset($page, $count) {
+    $page = ($page > 0) ? $page -= 1 : $page;
```

Comment:
This looks strange. In case when `$page > 0 == FALSE`, you just setting `$page` to `$page`, which looks rudimentary.
Would the following achieve what you wanted?
```php
if ($page < 1) {
  $page -= 1;
}
```